product_cost_security/models/product_template.py

```diff
@@ -7,3 +7,12 @@ class ProductTemplate(models.Model):
     _inherit = "product.template"
 
     standard_price = fields.Float(groups="product_cost_security.group_product_cost")
+    user_can_update_cost = fields.Boolean(compute="_compute_user_can_update_cost")
+
+    def _compute_user_can_update_cost(self):
```

Comment:
This compute should depend on the user:

```suggestion
    @api.depends_context("uid")
    def _compute_user_can_update_cost(self):
```